jupyterhub/app.py

```diff
@@ -1643,6 +1642,7 @@ def launch_instance_async(self, argv=None):
     @classmethod
     def launch_instance(cls, argv=None):
         self = cls.instance()
+        AsyncIOMainLoop().install()
```

Comment:
Not quite sure why this moves into the class method, other than its functionality changed with the deprecation.
http://www.tornadoweb.org/en/latest/asyncio.html#tornado.platform.asyncio.AsyncIOMainLoop

Reply:
This is a change I've been meaning to do for some time. I don't think triggering configuration at import time is generally the right thing to do. The fact that this was at the module level is the reason we've never been running tests with asyncio, despite all of the code in jupyterhub indicating that we are. Another bit of configuration in the notebook package was overriding that choice with pyzmq's eventloop at a later point, and thus taking precedence. Moving it here ensures that the choice takes place.

Reply:
Cool. Thanks for the explanation.